packages/visx-annotation/src/components/Label.tsx

```diff
@@ -100,6 +104,44 @@ export default function Label({
   const height = Math.floor(
     padding.top + padding.bottom + (titleBounds.height ?? 0) + (subtitleBounds.height ?? 0),
   );
+
+  const { wordsByLines: titleWordsByLine } = useText({
+    children: title,
+    verticalAnchor: 'start',
+    capHeight: titleFontSize,
+    fontSize: titleFontSize,
+    fontWeight: titleFontWeight,
+    fontFamily: titleProps?.fontFamily,
+    width: maxWidth,
+    ...titleProps,
+  });
+
+  const { wordsByLines: subtitleWordsByLine } = useText({
+    children: subtitle,
+    verticalAnchor: 'start',
+    capHeight: subtitleFontSize,
+    fontSize: subtitleFontSize,
+    fontWeight: subtitleFontWeight,
+    fontFamily: subtitleProps?.fontFamily,
+    width: maxWidth,
+    ...subtitleProps,
+  });
+
+  const titleMeasuredWidth = titleWordsByLine.reduce(
+    (maxTitleWidth, line) => Math.max(maxTitleWidth, line.width ?? 0),
+    0,
+  );
+
+  const subtitleMeasuredWidth = subtitleWordsByLine.reduce(
+    (maxSubtitleWidth, line) => Math.max(maxSubtitleWidth, line.width ?? 0),
+    0,
+  );
+
+  const textMeasuredWidth = Math.floor(
+    Math.min(maxWidth, Math.max(titleMeasuredWidth, subtitleMeasuredWidth)),
+  );
+  const measuredWidth = padding.right + padding.left + textMeasuredWidth;
+  const width = propWidth ?? measuredWidth;
```

Comment:
it's a bummer we can't skip the calculations above if `propWidth` **is** passed, but not easy to do with `hooks`. Theoretically could create a wrapper component explicitly to provide `width` if it's not passed, but I think we could consider that a future improvement if there are any perf issues. I didn't notice any lag in the annotation demo when quickly updating the title/subtitle text.

Reply:
Totally with you about this. I like this idea about component wrapper with measure logic within.